contrib/tally/handler.go

```diff
@@ -69,6 +69,36 @@ func NewMetricsHandler(scope tally.Scope) client.MetricsHandler {
 	return metricsHandler{scope}
 }
 
+// ScopeFromHandler returns the underlying scope of the handler. Callers may
+// need to check workflow.IsReplaying(ctx) to avoid recording metrics during
+// replay. If this handler was not created via this package, tally.NoopScope is
+// returned.
+//
+// Raw use of the scope is discouraged but may be used for Histograms or other
+// advanced features. This scope does not skip metrics during replay like the
+// metrics handler does. Therefore the caller should check replay state, for
+// example:
+// 	scope := tally.NoopScope
+// 	if !workflow.IsReplaying(ctx) {
+// 		scope = ScopeFromHandler(workflow.GetMetricsHandler(ctx))
+// 	}
+// 	scope.Histogram("my_histogram", nil).RecordDuration(5 * time.Second)
+func ScopeFromHandler(handler client.MetricsHandler) tally.Scope {
```

Comment:
if the handler is replay-aware, why would user need to check for IsReplaying flag?

Reply:
This handler is not replay aware. The replay aware handler wraps this one (and any metrics handler they may create, e.g. for OpenTelemetry). The raw scope is accessible at any time in replay or not.